internal/proxy/task_search.go

```diff
@@ -415,13 +415,14 @@ func (t *searchTask) searchShard(ctx context.Context, leaders []nodeInfo, channe
 		}
 		result, err := qn.Search(ctx, req)
 		if err != nil {
-			return err
-		}
-		if result.GetStatus().GetErrorCode() == commonpb.ErrorCode_NotShardLeader {
 			log.Warn("QueryNode search returns error", zap.Int64("nodeID", nodeID),
 				zap.Error(err))
 			return errInvalidShardLeaders
```

Comment:
It shouldn't returns errInvalidShardLeaders if the err is context.Canceled, it is fine to return this if err is grpc error or context.DeadlineExceeded

Reply:
Either way, proxy shall clean the cache and do search without cache.
Returning errInvalidShardLeaders is internal for proxy and just means: 
> Current shard leader cache is not working

Reply:
If just return the err, proxy shall check grpc err, errInvalidShardLeaders, context.Cancelled,  context.DeadlineExceed, which is just to redundant.